staging/src/k8s.io/apiserver/pkg/server/filters/goaway.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"math/rand"
+	"net/http"
+	"sync"
+)
+
+// GoawayDecider decides if server should send a GOAWAY
+type GoawayDecider interface {
+	Goaway(r *http.Request) bool
+}
+
+var (
+	// randPool used to get a rand.Rand and generate a random number thread-safely,
+	// which improve the performance of using rand.Rand with a locker
+	randPool = &sync.Pool{
+		New: func() interface{} {
+			return rand.New(rand.NewSource(1))
```

Comment:
You have to use a random seed, otherwise rejections will be correlated.

Reply:
I think it's not a big problem, as different P/goroutine get a same seed rand but invoke this function does not at the same time. 

Anyway I has changed it to be a random seed which may be more perfect, thanks a lot.

Reply:
Thanks for the fix; correlated random number streams are just a subtle source of bug that it's best to avoid.

E.g. if you imagine that all the apiservers end up with the same number of these in the pool, create at least some of them at the same time, and get about the same number of requests, then you'll end up with bursts where all the apiservers reject a request very near in time.

Reply:
yes, agreed